### PR TITLE
feat(setup): backup and unlink before link dotfiles

### PR DESCRIPTION
**Pull request by chengcyber**
# features

- backup current config files
- unlink current config files
- link dotfiles

# Note

Since I use macvim instead of neovim, I have not tested the specified cases on nvim configuration yet, but I wrote the logic to handle it.

### CODE REVIEW

File: setup/dotfiles.sh [hunk 1]
```diff
@@ -1,19 +1,70 @@
...
+vimrc
+vim
+)
+
+nvimrcpath="~/.config/nvim/init.vim"
+nvimpath="~/.config/nvimrc"
+
+fancy_echo "Backup current config"
+today=`date +%Y%m%d`
+# for i in ; do [ -e $i ] && [ ! -L $i ] && mv $i $i.$today; done
```
Comment (colbycheeze) [inline setup/dotfiles.sh]: Are these commented out lines needed?

File: setup/dotfiles.sh [hunk 2]
```diff
@@ -1,19 +1,70 @@
...
-ln -s ~/dotfiles/gitconfig ~/.gitconfig
-ln -s ~/dotfiles/gitmessage ~/.gitmessage
-ln -s ~/dotfiles/gvimrc ~/.gvimrc
-ln -s ~/dotfiles/tmux.conf ~/.tmux.conf
-ln -s ~/dotfiles/vimrc ~/.config/nvim/init.vim
-ln -s ~/.vim ~/.config/nvimrc
-ln -s ~/dotfiles/vimrc.bundles ~/.vimrc.bundles
-ln -s ~/dotfiles/zshenv ~/.zshenv
-ln -s ~/dotfiles/zshrc ~/.zshrc
+# ln -s ~/dotfiles/zsh ~/.zsh
```
Comment (colbycheeze) [inline setup/dotfiles.sh]: If the new code replaces this correctly, can go ahead and delete these commented out lines.

**Review by colbycheeze — CHANGES_REQUESTED** [review]
Cool! This looks helpful. To clarify that I am reading it correctly, does this loop through the list of dotfiles to link, back them up, then unlink existing links, and then do the linking process? (which would solve nested linkings if this script was ran twice for example)

Can you remove the commented out lines if they aren't needed, and I'll give it a merge.

File: setup/dotfiles.sh (hunk 1 above)
Comment (chengcyber) [reply setup/dotfiles.sh]: can be removed. The two is my current using logic, the code underneath have special logic to handle neovim.

File: setup/dotfiles.sh (hunk 2 above)
Comment (chengcyber) [reply setup/dotfiles.sh]: yes, you can . My only concern is test. Since Neovim is not my choice, and my dotfiles list is just vim/tmux/zsh.

**Comment by chengcyber** [comment]
updated: remove useless comment in dotfiles.sh.

I watched your sharing video on youtube, that's cool. Thanks for sharing, man

**Review by colbycheeze — APPROVED** [review]

**Pull request merged**